src/liboslexec/batched_backendllvm.cpp

```diff
@@ -813,37 +819,56 @@ BatchedBackendLLVM::llvm_load_mask(const Symbol& cond)
 llvm::Value*
 BatchedBackendLLVM::llvm_load_value(llvm::Value* ptr, const TypeSpec& type,
                                     int deriv, llvm::Value* arrayindex,
-                                    int component, TypeDesc cast,
-                                    bool op_is_uniform, bool index_is_uniform,
+                                    int component, bool ptr_is_uniform,
+                                    TypeDesc cast, bool op_is_uniform,
+                                    bool index_is_uniform,
                                     bool symbol_forced_boolean)
 {
     if (!ptr)
         return NULL;  // Error
 
+    TypeDesc t = type.simpletype();
+    llvm::Type *ptr_type, *ptr_scalar_type;
+
+    if (symbol_forced_boolean) {
+        if (ptr_is_uniform) {
+            ptr_type        = ll.type_bool();
+            ptr_scalar_type = ll.type_bool();
+        } else {
+            ptr_type        = ll.type_native_mask();
+            ptr_scalar_type = ll.type_native_mask();
```

Comment:
`  ptr_scalar_type = ll.type_native_mask();`
I'm not sure if that is quite right or not.  It just looked off to me.
Should it just be
`ptr_scalar_type = ll.type_bool();`
Perhaps its not used later when symbol_forced_bool && !ptr_is_uniform

Reply:
My understanding is that `ptr_type` and `ptr_scalar_type` are meant to have the same value for booleans, ints and floats. The "scalar" here refers to taking a single component of a triple or matrix.

Reply:
Hmm, I think scalar vs vector type when I see scalar.  Vs. element or component type.  So would `ptr_comp_type` be more accurate?

Reply:
Renamed now.